src/executables/csql_support.c

```diff
@@ -239,17 +239,97 @@ csql_invoke_system (const char *command)
     }
 }
 
+
+/*
+ * csql_invoke_formatter()
+ *   return: CSQL_SUCCESS/CSQL_FAILURE
+ *
+ * Note:
+ *   copy command editor buffer into temporary file and
+ *   invoke formatter. After the format is finished,
+ *   read the file into editor buffer.
+ */
+int
+csql_invoke_formatter ()
+{
+  auto[before_filename, before_fileptr] = filesys::open_temp_file ("bef_fmt_");
+  if (before_fileptr == NULL)
+    {
+      csql_Error_code = CSQL_ERR_OS_ERROR;
+      return CSQL_FAILURE;
+    }
+  filesys::auto_delete_file before_file_del (before_filename.c_str ());
+  filesys::auto_close_file before_file (before_fileptr);
+
+  if (csql_edit_write_file (before_file.get ()) == CSQL_FAILURE)
+    {
+      nonscr_display_error (csql_Scratch_text, SCRATCH_TEXT_LEN);
+      return CSQL_FAILURE;
+    }
+
+  auto[after_filename, after_fileptr] = filesys::open_temp_file ("aft_fmt_");
+  if (after_fileptr == NULL)
+    {
+      csql_Error_code = CSQL_ERR_OS_ERROR;
+      return CSQL_FAILURE;
+    }
+  filesys::auto_delete_file after_file_del (after_filename.c_str ());
+  filesys::auto_close_file after_file (after_fileptr);
+
+  char *cmd = csql_get_tmp_buf (strlen (csql_Formatter_cmd) + 1 + before_filename.size () + 1 + after_filename.size ());
```

Comment:
```suggestion
csql_invoke_formatter ()

int size;
char *cmd
.
.
.


size = strlen (csql_Formatter_cmd) + 1 + before_filename.size () + 1 + after_filename.size ();
  if ((cmd = csql_get_tmp_buf (size)) != NULL);
```

what about this code?

1. It is not easy to figure out what csql_get_tmp_buf has the argument.
2. Without initialized locally, I think variables should be initialized below the first paragraph.(Could you refer the legacy code?)

Reply:
Consistency to match legacy code.
Legacy codes for reference below.

```
char *cmd = csql_get_tmp_buf (strlen (csql_Editor_cmd) + 1 + filename.size ());
(csql_support.c)

char *cmd = csql_get_tmp_buf (1 + strlen (csql_Print_cmd) + 3 + filename.size ());
(csql.c)
```

Reply:
Ok. I see. Could you add comment above this line?
As you know, legacy code explains what it is.

`/* invoke the print command */`
`char *cmd = csql_get_tmp_buf (1 + strlen (csql_Print_cmd) + 3 + filename.size ()); (csql.c)`

Reply:
Ok. I'll add comment.
Thank you.